core/tee/tee_fs_common.c

```diff
@@ -1319,14 +1325,18 @@ int tee_fs_common_write(struct tee_fs_fd *fdp,
 int tee_fs_common_rename(const char *old, const char *new)
 {
 	int res = -1;
-	size_t old_len = strlen(old) + 1;
-	size_t new_len = strlen(old) + 1;
+	size_t old_len;
+	size_t new_len;
 	struct tee_fs_dir *old_dir;
 	struct tee_fs_dirent *dirent;
 	char *meta_filename = NULL;
 
+	TEE_ASSERT(old != NULL && new != NULL);
```

Comment:
Sure, it hangs the system before the abort can occur. I understand the it helps while debugging, but on a production system you would be much better of with a returned error.
Here's an interesting link on the matter http://yarchive.net/comp/linux/BUG.html

By the way, I don't think all TEE_ASSERT should be removed now, we should just be careful when new ones as they will bring the system down if triggered.